src/main/kotlin/logic/feature/projectHistoryHandling/HistoryService.kt

```diff
@@ -0,0 +1,28 @@
+package org.damascus.logic.feature.projectHistoryHandling
+
+import org.damascus.logic.entities.ActionLog
+import org.damascus.logic.repositories.HistoryRepository
+import org.damascus.utiles.NoHistoryException
+import java.util.UUID
+
+class HistoryService(
+    private val historyRepository: HistoryRepository
+) {
+    fun saveLog(actionLog: ActionLog){
+        historyRepository.saveLog(actionLog)
+    }
+
+    fun getAllLogs() : List<ActionLog>{
+        val logs = historyRepository.getAllLogs()
+        if (logs.isEmpty()) {
+            throw NoHistoryException("No logs found")
+        }
+        return logs
+    }
```

Comment:
What purpose (use-case) does this function serve?

Reply:
Yes, I can understand that, you used a good name for the function.

My question is, when will this be used within our system given its requirements?

The history feature is shown for the entire project, you added get `getLogByTaskId` I thought it was a nice addition if we wanted to make it an extra feature, however, the get all logs for all projects, doesn't seem to serve a certain use-case.